particles/Processing/js/ExpandDims.js

```diff
@@ -8,20 +8,18 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-defineParticle(({DomParticle, log}) => {
+defineParticle(({DomParticle, resolver, log}) => {
+
+  importScripts(resolver(`$here/tf.js`));
 
   return class extends DomParticle {
-    update({resource}) {
-      if (resource) {
-        this.apply(resource);
+    async update({tensor, axis}) {
+      if (tensor && axis) {
+        log('Expanding Dimensions...');
+        await this.set('newTensor', await this.tf.expandDims(tensor, axis));
```

Comment:
This is really clean. I like it a lot!

I wasn't aware that we could just `set` handles. This makes things really clear and simple.

Reply:
One issue: `newTensor`, to the best of my knowedge, should be "boxed", e.g. 
```
await this.set('newTensor', {ref: await this.tf.expandDims(tensor, axis)});
```

Reply:
Just checked. You take care of this in the `tf.js`. Brilliant.

Reply:
‘set’ is experimental api I put into tf.js as a try-out. See notes in that file.